api/v3/Activity/Getdayswithactivities.php

```diff
@@ -48,10 +52,17 @@ function civicrm_api3_activity_getdayswithactivities(array $params) {
   }
 
   if (!empty($params['case_id'])) {
+    print('<pre>' . print_r($params['case_id'], TRUE) . '</pre>');
+
     $query->join('ca', "INNER JOIN civicrm_case_activity AS ca ON a.id = ca.activity_id");
     _civicrm_api3_activity_getdayswithactivities_handle_id_param($params['case_id'], 'ca.case_id', $query);
   }
 
+  if (!empty($params['case_params'])) {
+    $query->join('ca', "INNER JOIN civicrm_case_activity AS ca ON a.id = ca.activity_id");
+    _civicrm_api3_activity_getdayswithactivities_handle_id_param(['IN' => _get_case_ids($params['case_params'])], 'ca.case_id', $query);
```

Comment:
Can we check the return values of `_get_case_ids($params['case_params']` first before using ? It might turn out to be empty in rare cases and would cause problems with the SQL query.

Reply:
Since the case_id and case_params parameters both require joins to the case activity table. I was thinking we can extract this to a function and use for both. e.g

```php
function _join_to_case($query, $value) {
    $query->join('ca', "INNER JOIN civicrm_case_activity AS ca ON a.id = ca.activity_id");
    _civicrm_api3_activity_getdayswithactivities_handle_id_param($value, 'ca.case_id', $query);
}
```

Reply:
As discussed we can change this parameter name to `case_filter` for consistency.